src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java

```diff
@@ -910,6 +956,78 @@ private String getJsonPath(TangibleSubResource subResource, String pathSuffix) {
         return TANGIBLE_NOTE + "." + subResource.getJsonPath() + pathSuffix;
     }
 
+    @Override
+    public Errors crossValidate(Errors errors, HttpServletRequest request, String companyAccountsId,
+            TangibleAssets tangibleAssets) throws DataException {
+
+        crossValidateCurrentPeriod(errors, request, companyAccountsId, tangibleAssets);
+        crossValidatePreviousPeriod(errors, request, companyAccountsId, tangibleAssets);
+
+        return errors;
+    }
+
+    private void crossValidateCurrentPeriod(Errors errors, HttpServletRequest request, String companyAccountsId,
+            TangibleAssets tangibleAssets) throws DataException {
+
+        String currentPeriodId = currentPeriodService.generateID(companyAccountsId);
+
+        ResponseObject<CurrentPeriod> currentPeriodResponseObject =
+                currentPeriodService.findById(currentPeriodId, request);
+        CurrentPeriod currentPeriod = currentPeriodResponseObject.getData();
+
+        Long currentPeriodTangible =
+                Optional.ofNullable(currentPeriod)
```

Comment:
We are using a combination of both lambda's and method references to do the same thing could we change it to use one or the other not both in the same file.